### PR TITLE
status: correct data in non-root

### PR DESCRIPTION
**Pull request by CalvoM**
## Proposed Commit Message

status: correct data in non-root

Currently, when `ua status` as non-root it gets its status from status cache file.
Incase of any changes on the contract server concerning the machine's account,
we do not get up to date info.

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/tests/test_status.py
Comment (lucasmoura) [inline uaclient/tests/test_status.py:579]: Do we still need this mock here ?

File: uaclient/tests/test_status.py
Comment (lucasmoura) [inline uaclient/tests/test_status.py:581]: I think we also need a check to see if `root` and `non-root` status are the same

File: uaclient/tests/test_status.py
Comment (lucasmoura) [inline uaclient/tests/test_status.py:809]: Why do we need this fixture here ?

File: uaclient/tests/test_status.py
Comment (lucasmoura) [inline uaclient/tests/test_status.py:878]: Shouldn't they be the same here ?

File: uaclient/tests/test_status.py
Comment (CalvoM) [reply uaclient/tests/test_status.py:579]: No we do not actually. Removing.

File: uaclient/tests/test_status.py
Comment (CalvoM) [reply uaclient/tests/test_status.py:809]: I was testing something, forgot to cleanup.

File: uaclient/tests/test_status.py
Comment (CalvoM) [reply uaclient/tests/test_status.py:878]: No, because the `expected_status` is written to status-cache which non-root does not use.

File: uaclient/tests/test_status.py
Comment (CalvoM) [reply uaclient/tests/test_status.py:569]: Makes sense.

**Review by lucasmoura — COMMENTED** [review]
Maybe I missing this test, but couldn't spot one where attached non-root and root report the same values. We could use this test to also assert than even on attached scenarios, we don't use the `status-cache`

File: uaclient/tests/test_status.py
Comment (CalvoM) [reply uaclient/tests/test_status.py:575]: Actually, I found out that it is the same `tmpdir` object.

**Review by renanrodrigo — APPROVED** [review]
lgtm, pending only Lucas's review points (test the `try`s)

File: uaclient/entitlements/fips.py
Comment (lucasmoura) [inline uaclient/entitlements/fips.py:322]: Let's add a test here that shows that a non-root call where we should delete the notices, doesn't fail the `application_status` call

**Review by lucasmoura — COMMENTED** [review]
@CalvoM just noticed that the `machine_token.json` file is now being stored outside of the `private` folder under `/var/lib/ubuntu-advantage/`. This looks like a bug on the `MachineTokenFile` class.

Additionally, I performed an extra test where I deleted the status cache on an attached machine and ran `pro status` as non-root. In that situation, the output is showing the system as unattached. This looks odd, since I believe we shouldn't be relying on the status cache for status anymore

File: uaclient/entitlements/fips.py
Comment (orndorffgrant) [inline uaclient/entitlements/fips.py:308]: I'd rather see this function on the notice_file itself, so it can be used in other places if needed.

It could look like this:
```python
notice_file.remove() # raises NonRootUserError when not root
notice_file.try_remove() # logs warning but doesn't raise error when not root
```

Comment (orndorffgrant) [reply uaclient/entitlements/fips.py:308]: for example, I think we need to catch the NonRootUserError in cli.py line 1557

File: uaclient/tests/test_status.py
Comment (orndorffgrant) [inline uaclient/tests/test_status.py:542]: I think this whole test can just be removed, because its purpose was to test that the nonroot used the cache. Right? @lucasmoura @CalvoM

File: uaclient/tests/test_status.py
Comment (orndorffgrant) [inline uaclient/tests/test_status.py:593]: We should have a version of this test for an attached config.

File: uaclient/tests/test_status.py
Comment (orndorffgrant) [reply uaclient/tests/test_status.py:575]: the tmpdir is created when creating the fixture. So each unit test gets its own tmpdir, but within a function, each instance of `FakeConfig` will re-use the same tmpdir

File: uaclient/tests/test_status.py
Comment (orndorffgrant) [inline uaclient/tests/test_status.py:821]: `for_attached_machine` is a classmethod
```suggestion
        cfg = FakeConfig.for_attached_machine(
```

File: uaclient/tests/test_status.py
Comment (orndorffgrant) [reply uaclient/tests/test_status.py:868]: (Accidentally clicked the resolve/unresolve button three times haha)

If we don't expect the status to be `expected_status` then `expected_status` isn't really the expected status, is it?

Let's rename `expected_status` to `status_cache_value`

**Review by renanrodrigo — APPROVED** [review]
lgtm

File: uaclient/tests/test_status.py
Comment (orndorffgrant) [reply uaclient/tests/test_status.py:828]: @CalvoM tagging to make sure you see this comment ^^

**Review by orndorffgrant — APPROVED** [review]
Looks good! Please just see my suggestion about renaming the non-expected `expected_status` test variable

File: uaclient/tests/test_status.py
Comment (CalvoM) [reply uaclient/tests/test_status.py:828]: I deleted the test as we do not use `status_cache` for non-root user. @orndorffgrant

**Comment by orndorffgrant** [comment]
Thanks for the small fix @CalvoM LGTM

@lucasmoura did you want to look at this again before we merge? (since you've left several comments on it so far)

**Pull request merged**
